### ISSUE TITLE
Phone number getting parsed as a date

### ISSUE THREAD

**Issue by abhishek-butola** [UgnisSoftware/react-spreadsheet-import#119]
Have a field called phone number in this format 564-564-5644.

After getting parsed it is replaced with a date string 1900-01-00.

This only happening for .csv files if i convert csv to xlsx then it is showing correct result.

**Comment by masiulis** [comment]
Hey @abhishek-butola, are you passing any `dateFormat` prop to the component?

For reference, parsing is done by `xlxs` library here [https://github.com/UgnisSoftware/react-spreadsheet-import/blob/master/src/steps/UploadStep/components/DropZone.tsx#L42](https://github.com/UgnisSoftware/react-spreadsheet-import/blob/master/src/steps/UploadStep/components/DropZone.tsx#L42) and transformation to json here [https://github.com/UgnisSoftware/react-spreadsheet-import/blob/master/src/utils/mapWorkbook.ts](https://github.com/UgnisSoftware/react-spreadsheet-import/blob/master/src/utils/mapWorkbook.ts).

**Comment by abhishek-butola** [comment]
![image](https://user-images.githubusercontent.com/27547546/178475412-392e9564-a5f5-4148-86d0-3656c34f60e4.png)
CSV file giving incorrect phone

**Comment by abhishek-butola** [comment]
![image](https://user-images.githubusercontent.com/27547546/178475559-26a74289-382a-4b4e-9d36-cedd8e15b03c.png)
XLSX gives correct

**Comment by abhishek-butola** [comment]
`      const workbook = XLSX.read(arrayBuffer, dateFormat ? { cellDates: true, dateNF: dateFormat } : undefined)`

This could fix. But what if we have both phone and date column.

**Comment by abhishek-butola** [comment]
Tried manually parsing the csv. It fixed my usecase of invalid phone. But still some of tests are failing

```
//readFilesAsync.ts
export const readFileAsync = (file: File) => {
  return new Promise((resolve, reject) => {
    const reader = new FileReader()

    reader.onload = () => {
      resolve(reader.result)
    }

    reader.onerror = reject

    if (file.type === "text/csv") {
      reader.readAsText(file)
    } else {
      reader.readAsArrayBuffer(file)
    }
  })
}
```

```
//getFileBuffer.ts
import csvToJson from "csvtojson"
import * as XLSX from "xlsx"
import { readFileAsync } from "./readFilesAsync"

export const getFileBuffer = async (file: File) => {
  let buffer
  if (file.type === "text/csv") {
    const csvString = await readFileAsync(file)
    const formattedJson = await csvToJson().fromString(csvString as string)

    const workbook: XLSX.WorkBook = XLSX.utils.book_new()
    const worksheet: XLSX.WorkSheet = XLSX.utils.json_to_sheet(formattedJson)
    XLSX.utils.book_append_sheet(workbook, worksheet)
    buffer = XLSX.write(workbook, { type: "buffer" })
  } else {
    buffer = await readFileAsync(file)
  }
  return buffer
}

```

//DropZone.tsx
```
  onDrop: async ([file]) => {
      setLoading(true)
      const buffer = await getFileBuffer(file)
      const workbook = XLSX.read(buffer, { cellDates: true, dateNF: dateFormat })
      setLoading(false)
      onContinue(workbook)
    },
```
![image](https://user-images.githubusercontent.com/27547546/178700037-ec62d2b0-b284-46e0-8e16-ea3b2fb2e38a.png)

**Comment by abhishek-butola** [comment]
#121 added parseRaw parameter for XLSX

**Comment by JulitorK** [comment]
thank you @abhishek-butola,  #121 merged

**Comment by jamesicon** [comment]
Any idea when this will be released? Trying to use this in production but CSV is malforming SSN and phone numbers

**Comment by JulitorK** [comment]
Hi, @jamesicon, released with 2.0.8. 
If you run into more issues with formatting, please check this [comment](https://github.com/UgnisSoftware/react-spreadsheet-import/pull/121#issuecomment-1194124702), maybe it will be of use.
